jwst/refpix/irs2_subtract_reference.py

```diff
@@ -387,6 +386,86 @@ def decode_mask(output, mask):
     return bits
 
 
+def rm_intermittent_badpix(data, scipix_n, refpix_r):
+    """Find pixels that are intermittently bad and replace with nearest
+    good value.
+
+    Parameters
+    ----------
+    data : 4-D ndarray
+        The data array in detector orientation.  This includes both the
+        science and interleaved reference pixel values.  `data` will be
+        modified in-place. The science data values will not be modified.
+
+    scipix_n : int
+        Number of regular (science) samples before stepping out to collect
+        reference samples.
+
+    refpix_r : int
+        Number of reference samples before stepping back in to collect
+        regular samples.
+
+    Returns
+    -------
+    data : 4-D ndarray
+        The data array in detector orientation.  This includes both the
+        science and interleaved reference pixel values.  The intermittently
+        bad pixels are now set to the nearest good reference pixel value.
+    """
+    # The intermittently bad pixels will be replaced for all integrations
+    # and all groups. The last group will be used to identify them
+    nits, ngroups, ny, nx = np.shape(data)
+    total_rp2replace = []
+    # calculate differences of even and odd pairs per amplifier
+    amplifier = nx // 5                 # 640
+    for k in range(1, 5):
+        diffs, rp2check = [], []
+        offset = int(k * amplifier)
+        # jump from the start of the reference pixel sequence to the next
+        ref_pix, rp2check, pair = [], [], 0
+        for rpstart in range(scipix_n//2, amplifier, scipix_n+refpix_r):
+            rpstart += offset
+            # go through the 4 reference pixels
+            for ri in range(4):
+                ri = rpstart + ri
+                ref_pix.append(ri)
+                rp_val = np.mean(data[nits-1, ngroups-1, :, ri])
```

Comment:
I see more uses of "nits" here, so I guess the code is at least consistent. But all the rest of the pipeline code base uses "nints" for the number of integrations, so for consistency with the rest of the pipeline I think it would be good to change all of these to "nints".

Reply:
I made it consistent with above